src/services/index.ts

```diff
@@ -0,0 +1,448 @@
+       && token_name
+       && token_symbol) {
+      if (!token_name || !token_symbol) {
+        logger.error('Processed a token creation event but it did not come with token name and symbol');
+        process.exit(1);
+      }
+      await storeTokenInformation(mysql, hash, token_name, token_symbol);
+    }
+
+    // check if any of the inputs are still marked as locked and update tables accordingly.
+    // See remarks on getLockedUtxoFromInputs for more explanation. It's important to perform this
+    // before updating the balances
+    const lockedInputs = await getLockedUtxoFromInputs(mysql, inputs);
+    await unlockUtxos(mysql, lockedInputs, true);
+
+    // add transaction outputs to the tx_outputs table
+    markLockedOutputs(txOutputs, now, heightlock !== null);
+
+    // Add the transaction
+    logger.debug('Will add the tx with height', height);
+    await addOrUpdateTx(
+      mysql,
+      hash,
+      height,
+      timestamp,
+      version,
+      weight,
+    );
+
+    // Add utxos
+    await addUtxos(mysql, hash, txOutputs, heightlock);
+    await updateTxOutputSpentBy(mysql, txInputs, hash);
+
+    // Handle genesis parent txs:
+    if (inputs.length > 0 || outputs.length > 0) {
+      const tokenList: string[] = getTokenListFromInputsAndOutputs(txInputs, txOutputs);
+
+      // Update transaction count with the new tx
+      await incrementTokensTxCount(mysql, tokenList);
+
+      const addressBalanceMap: StringMap<TokenBalanceMap> = getAddressBalanceMap(txInputs, txOutputs);
+
+      // update address tables (address, address_balance, address_tx_history)
+      await updateAddressTablesWithTx(mysql, hash, timestamp, addressBalanceMap);
+
+      // for the addresses present on the tx, check if there are any wallets associated
+      const addressWalletMap: StringMap<Wallet> = await getAddressWalletInfo(mysql, Object.keys(addressBalanceMap));
+
+      // for each already started wallet, update databases
+      const seenWallets = new Set();
+      for (const wallet of Object.values(addressWalletMap)) {
+        const walletId = wallet.walletId;
+
+        // this map might contain duplicate wallet values, as 2 different addresses might belong to the same wallet
+        if (seenWallets.has(walletId)) continue;
+        seenWallets.add(walletId);
+        const { newAddresses, lastUsedAddressIndex } = await generateAddresses(mysql, wallet.xpubkey, wallet.maxGap);
+        // might need to generate new addresses to keep maxGap
+        await addNewAddresses(mysql, walletId, newAddresses, lastUsedAddressIndex);
+        // update existing addresses' walletId and index
+      }
+      // update wallet_balance and wallet_tx_history tables
+      const walletBalanceMap: StringMap<TokenBalanceMap> = getWalletBalanceMap(addressWalletMap, addressBalanceMap);
+      await updateWalletTablesWithTx(mysql, hash, timestamp, walletBalanceMap);
+    }
+
+    // TODO: Send message on SQS  for real-time update
```

Comment:
Should this be done in this PR?

Reply:
I would rather do it after the monorepo merge as this also requires some changes in the wallet-service